### PR TITLE
chore: update eslint, using our shared conf, and update every dev deps

### PR DESCRIPTION
**Pull request by edodusi**
## Pull request type

Package update, linting refactoring

- [ ] Bugfix
- [ ] Feature
- [X] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Other (please describe): deps update

## How to test this PR

Update the dependency in any running project: nothing should break.

## What is the new behavior?

Style changes.

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { storyblokLintConfig } from '@storyblok/eslint-config';
+
+export default storyblokLintConfig({
+  rules: {
+    '@typescript-eslint/no-this-alias': 'off',
```
Comment (alexjoverm) [inline eslint.config.mjs:5]: @edodusi Are these rules definite, or temporary? If they're temporary, I suggest we add to-do comment, on the lines of `// @TODO: remove after fixing in v7`

**Review by alexjoverm — CHANGES_REQUESTED** [review]
Hey @edodusi, thanks for this PR! A few small changes left to do after reviewing the internal checklist of things to update:

- Update to Node LTS [here](https://github.com/storyblok/storyblok-js-client/blob/main/.github/workflows/release.yml#L24), [here](https://github.com/storyblok/storyblok-js-client/blob/main/.github/workflows/test.yml#L18-L21), [here](https://github.com/storyblok/storyblok-js-client/blob/main/.github/workflows/build.yml#L18-L21) and [here](https://github.com/storyblok/storyblok-js-client/blob/main/.github/workflows/pkg.pr.new.yml#L27-L29) (if somehow the version can be unified in a single place, that'd be amazing)
- Update to latest Semantic Release https://github.com/storyblok/storyblok-js-client/blob/main/.github/workflows/release.yml#L32

Also in the ticket we defined to remove husky and lint-staged, towards the approach of having a `lint` job in the CI doing this, but I don't see the change here. Is it on track or there is a particular reason not to have it?

**Review by alvarosabu — CHANGES_REQUESTED** [review]
@edodusi  Since adding the Lint CI action is part of the strategy of replacing the pre-push husky one I would include it on the scope of this PR. Check https://github.com/storyblok/storyblok-nuxt/commit/a22f89b544869eb6afcc017173ed42cdc195b80f for inspo

File: eslint.config.mjs (hunk 1 above)
Comment (edodusi) [reply eslint.config.mjs:5]: they're temporary, will add the comment!

**Comment by edodusi** [comment]
@alexjoverm @alvarosabu changes made 👀

**Review by alexjoverm — APPROVED** [review]

**Pull request merged**